### DISCUSSION TITLE
NAPALM Error

### DISCUSSION THREAD

**Discussion by kopacko** [netbox-community/netbox#9572] [General]
I was attempting to get NAPALM up and running on my new NetBox instance running v3.2.4 and I get this error in the logs:
```
django.core.exceptions.ImproperlyConfigured: Plugin napalm does not provide a 'config' variable.
This should be defined in the plugin's __init__.py file and point to the PluginConfig subclass.
```
Following the documents I added this to **configuration.py**:
```
# Enable installed plugins. Add the name of each plugin to the list.
PLUGINS = [ 'napalm' ]
```
I assume I probably missed something. I had it working for a moment and restarted the box and now it doesn't work.

**Comment by joaolucasmacedo** [comment]
You don't need do declare NAPALM as a plugin. Just install tha packages on virtual environment, set login and password at the configuration.py file and the driver name on "Platforms".

**Comment by kopacko** [comment]
@joaolucasmacedo That worked. Thank you!

I guess the NetBox docs need to be updated then?

**Reply by kkthxbye-code** [reply]
> I guess the NetBox docs need to be updated then?

Why? I just looked and I can't find where it tells you to add 'napalm' to the plugins list.

**Reply by kopacko** [reply]
Yea you better "edit" that post and be a little less aggressive next time. I did not do as you said:
`Why? I just looked and I can't find where it tells you to add 'napalm' to the plugins list. I think you came up with that on your own.`

This is where I referenced adding it to the plugins list:
![image](https://user-images.githubusercontent.com/1976185/174831910-791082c9-ed56-4d98-a796-71c25c7b687e.png)

I did this:
```
$ source /opt/netbox/venv/bin/activate
(venv) $ pip install napalm
```
Then did this:
```
PLUGINS = [
    'napalm',
]
```

**Reply by kkthxbye-code** [reply]
That documentation is for plugins and doesn't mention napalm at all.

**Reply by kopacko** [reply]
Yeap, I had just learned that from @joaolucasmacedo 's reply. But that whole subsequent sentence `I think you came up with that on your own.` Is where I replied with how I got to where I was the first time.

**Reply by kkthxbye-code** [reply]
So the documentation is fine? What are we talking about then? I'm not sure I follow.

**Reply by kopacko** [reply]
Yes the documentation is fine. I was working under the assumption that NAPALM was a plugin.

**Reply by ketchup57** [reply]
nope just a terrible reference.
